src/program.cpp

```diff
@@ -167,13 +168,37 @@ target_assignments program::get_target_assignments(const std::vector<target>& ta
     target_assignments p;
 
     const auto* mod = get_main_module();
-    for(auto it : iterator_for(*mod))
+    std::vector<supported_segments> target_subgraphs;
+    target_subgraphs.reserve(targets.size());
+    std::transform(targets.begin(),
+                   targets.end(),
+                   std::back_inserter(target_subgraphs),
+                   [mod, m](const auto& t) { return t.find_supported(mod, m); });
```

Comment:
Use `[&]` for capture. And also return `make_pair(t, t.find_supported(mod, m))`.